### PR TITLE
Do not do string concatenation on logger, except where guarded with i…

### PR DESCRIPTION
**Pull request by StrongestNumber9**
…sTraceEnabled

### CODE REVIEW

File: src/main/java/com/teragrep/rlo_12/DirectoryEventWatcher.java [hunk 1]
```diff
@@ -303,24 +308,24 @@ public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOEx
 
         @Override
         public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
-            LOGGER.trace("preVisitDirectory <[" + dir.toAbsolutePath() + "]>");
+            LOGGER.trace("preVisitDirectory <[{}]>", dir.toAbsolutePath());
             if (dir.toAbsolutePath().equals(initialDirectory) || recursive) {
                 if (!dir.toFile().canRead()) {
                     // non-readables are skipped
-                    LOGGER.warn("Directory <[" + dir.toAbsolutePath() + "]> is not readable, skipping.");
+                    LOGGER.warn("Directory <[{}]> is not readable, skipping.", dir.toAbsolutePath());
```
Comment (kortemik) [inline src/main/java/com/teragrep/rlo_12/DirectoryEventWatcher.java:315]: dir.toAbsolutePath() is a method call, therefore must be guarded

File: src/main/java/com/teragrep/rlo_12/DirectoryEventWatcher.java [hunk 2]
```diff
@@ -303,24 +308,24 @@ public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOEx
...
                     return FileVisitResult.SKIP_SUBTREE;
                 }
                 register(dir.toAbsolutePath());
             } else {
-                LOGGER.trace("Directory skipped <[" + dir.toAbsolutePath() + "]> due to non-recursive processing");
+                LOGGER.trace("Directory skipped <[{}]> due to non-recursive processing", dir.toAbsolutePath());
```
Comment (kortemik) [inline src/main/java/com/teragrep/rlo_12/DirectoryEventWatcher.java:320]: dir.toAbsolutePath() is a method call, therefore must be guarded

File: src/main/java/com/teragrep/rlo_12/FileStatusManager.java [hunk 3]
```diff
@@ -64,8 +64,9 @@ public void run() {
                         }
                         // read it
                         pathCurrentStateMap.put(changedFile.getPath(), MonitoredFile.Status.MODIFIED_SYNCHRONIZING);
-                        LOGGER.trace("Path <[" + changedFile.getPath() + "]> state changed from <NONE> to <" + MonitoredFile.Status.MODIFIED_SYNCHRONIZING + "> because <" + changedFile.getStatus() + ">");
-
+                        if(LOGGER.isTraceEnabled()) {
+                            LOGGER.trace("Path <[" + changedFile.getPath() + "]> state changed from <NONE> to <" + MonitoredFile.Status.MODIFIED_SYNCHRONIZING + "> because <" + changedFile.getStatus() + ">");
```
Comment (kortemik) [inline src/main/java/com/teragrep/rlo_12/FileStatusManager.java:68]: change to formatter pattern

File: src/main/java/com/teragrep/rlo_12/FileStatusManager.java [hunk 4]
```diff
@@ -77,23 +78,31 @@ public void run() {
                             switch (currentState) {
                                 case MODIFIED_SYNCHRONIZED:
                                     pathCurrentStateMap.put(changedFile.getPath(), MonitoredFile.Status.MODIFIED_SYNCHRONIZING);
-                                    LOGGER.trace("Path <[" + changedFile.getPath() + "]> state changed from <" + currentState + "> to <" + MonitoredFile.Status.MODIFIED_SYNCHRONIZING + "> because <" + changedFile.getStatus() + ">");
+                                    if(LOGGER.isTraceEnabled()) {
+                                        LOGGER.trace("Path <[" + changedFile.getPath() + "]> state changed from <" + currentState + "> to <" + MonitoredFile.Status.MODIFIED_SYNCHRONIZING + "> because <" + changedFile.getStatus() + ">");
```
Comment (kortemik) [inline src/main/java/com/teragrep/rlo_12/FileStatusManager.java:82]: change to formatter pattern

File: src/main/java/com/teragrep/rlo_12/FileStatusManager.java [hunk 5]
```diff
@@ -77,23 +78,31 @@ public void run() {
...
+                                    }
                                     threadPoolExecutor.execute(new FileTask(transferQueue, new MonitoredFile(changedFile.getPath(), MonitoredFile.Status.SYNC_MODIFIED), consumerSupplier.get()));
                                     break;
                                 case MODIFIED_SYNCHRONIZING:
                                     // still being read, not resubmitting, but indicating here that it needs to be re-read once reading completes
                                     pathCurrentStateMap.put(changedFile.getPath(), MonitoredFile.Status.SYNC_MODIFIED);
-                                    LOGGER.trace("Path <[" + changedFile.getPath() + "]> state changed from <" + currentState + "> to <" + MonitoredFile.Status.SYNC_MODIFIED + "> because <" + changedFile.getStatus() + "> (modified while reading)");
+                                    if(LOGGER.isTraceEnabled()) {
+                                        LOGGER.trace("Path <[" + changedFile.getPath() + "]> state changed from <" + currentState + "> to <" + MonitoredFile.Status.SYNC_MODIFIED + "> because <" + changedFile.getStatus() + "> (modified while reading)");
```
Comment (kortemik) [inline src/main/java/com/teragrep/rlo_12/FileStatusManager.java:90]: change to formatter pattern

File: src/main/java/com/teragrep/rlo_12/FileStatusManager.java [hunk 6]
```diff
@@ -77,23 +78,31 @@ public void run() {
...
+                                    }
                                     break;
                                 case SYNC_DELETED:
                                 case DELETE_SYNCHRONIZING:
                                     pathCurrentStateMap.put(changedFile.getPath(), MonitoredFile.Status.SYNC_RECREATED);
-                                    LOGGER.trace("Path <[" + changedFile.getPath() + "]> state changed from <" + currentState + "> to <" + MonitoredFile.Status.SYNC_RECREATED + "> because <" + changedFile.getStatus() + ">");
+                                    if(LOGGER.isTraceEnabled()) {
+                                        LOGGER.trace("Path <[" + changedFile.getPath() + "]> state changed from <" + currentState + "> to <" + MonitoredFile.Status.SYNC_RECREATED + "> because <" + changedFile.getStatus() + ">");
```
Comment (kortemik) [inline src/main/java/com/teragrep/rlo_12/FileStatusManager.java:97]: change to formatter pattern

File: src/main/java/com/teragrep/rlo_12/FileStatusManager.java [hunk 7]
```diff
@@ -77,23 +78,31 @@ public void run() {
...
+                                    }
                                     break;
                                 case SYNC_MODIFIED:
                                 case SYNC_RECREATED:
                                     // no-op
-                                    LOGGER.trace("Path <[" + changedFile.getPath() + "]> state not changed from <" + currentState + "> because <" + changedFile.getStatus() + ">");
+                                    if(LOGGER.isTraceEnabled()) {
+                                        LOGGER.trace("Path <[" + changedFile.getPath() + "]> state not changed from <" + currentState + "> because <" + changedFile.getStatus() + ">");
```
Comment (kortemik) [inline src/main/java/com/teragrep/rlo_12/FileStatusManager.java:104]: change to formatter pattern

File: src/main/java/com/teragrep/rlo_12/FileStatusManager.java [hunk 8]
```diff
@@ -106,15 +115,19 @@ public void run() {
                             switch (currentState) {
                                 case MODIFIED_SYNCHRONIZED:
                                     // removed when everything (to our understanding) was read
-                                    LOGGER.trace("Path <[" + changedFile.getPath() + "]> state changed from <" + currentState + "> to <" + MonitoredFile.Status.DELETE_SYNCHRONIZING + "> because <" + changedFile.getStatus() + ">");
+                                    if(LOGGER.isTraceEnabled()) {
+                                        LOGGER.trace("Path <[" + changedFile.getPath() + "]> state changed from <" + currentState + "> to <" + MonitoredFile.Status.DELETE_SYNCHRONIZING + "> because <" + changedFile.getStatus() + ">");
```
Comment (kortemik) [inline src/main/java/com/teragrep/rlo_12/FileStatusManager.java:119]: change to formatter pattern

File: src/main/java/com/teragrep/rlo_12/FileStatusManager.java [hunk 9]
```diff
@@ -106,15 +115,19 @@ public void run() {
...
                                     pathCurrentStateMap.put(changedFile.getPath(), MonitoredFile.Status.DELETE_SYNCHRONIZING);
                                     threadPoolExecutor.execute(new FileTask(transferQueue, new MonitoredFile(changedFile.getPath(), MonitoredFile.Status.SYNC_DELETED), consumerSupplier.get()));
                                     break;
                                 case SYNC_RECREATED:
                                 case SYNC_MODIFIED:
                                 case MODIFIED_SYNCHRONIZING:
                                     // task is already on this, indicate that it was removed
-                                    LOGGER.trace("Path <[" + changedFile.getPath() + "]> state changed from <" + currentState + "> to <" + MonitoredFile.Status.SYNC_DELETED + "> because <" + changedFile.getStatus() + ">");
+                                    if(LOGGER.isTraceEnabled()) {
+                                        LOGGER.trace("Path <[" + changedFile.getPath() + "]> state changed from <" + currentState + "> to <" + MonitoredFile.Status.SYNC_DELETED + "> because <" + changedFile.getStatus() + ">");
```
Comment (kortemik) [inline src/main/java/com/teragrep/rlo_12/FileStatusManager.java:129]: change to formatter pattern

File: src/main/java/com/teragrep/rlo_12/FileStatusManager.java [hunk 10]
```diff
@@ -132,20 +145,28 @@ public void run() {
                                 case MODIFIED_SYNCHRONIZING:
                                     // file has been read and not modified since reading started
                                     pathCurrentStateMap.put(changedFile.getPath(), MonitoredFile.Status.MODIFIED_SYNCHRONIZED);
-                                    LOGGER.trace("Path <[" + changedFile.getPath() + "]> state changed from <" + currentState + "> to <" + MonitoredFile.Status.MODIFIED_SYNCHRONIZED + "> because <" + changedFile.getStatus() + ">");
+                                    if(LOGGER.isTraceEnabled()) {
+                                        LOGGER.trace("Path <[" + changedFile.getPath() + "]> state changed from <" + currentState + "> to <" + MonitoredFile.Status.MODIFIED_SYNCHRONIZED + "> because <" + changedFile.getStatus() + ">");
```
Comment (kortemik) [inline src/main/java/com/teragrep/rlo_12/FileStatusManager.java:149]: change to formatter pattern

File: src/main/java/com/teragrep/rlo_12/FileStatusManager.java [hunk 11]
```diff
@@ -132,20 +145,28 @@ public void run() {
...
+                                    }
                                     break;
                                 case SYNC_RECREATED:
-                                    LOGGER.trace("Path <[" + changedFile.getPath() + "]> state changed from <" + currentState + "> to <" + MonitoredFile.Status.MODIFIED_SYNCHRONIZING + "> because <" + changedFile.getStatus() + ">");
+                                    if(LOGGER.isTraceEnabled()) {
+                                        LOGGER.trace("Path <[" + changedFile.getPath() + "]> state changed from <" + currentState + "> to <" + MonitoredFile.Status.MODIFIED_SYNCHRONIZING + "> because <" + changedFile.getStatus() + ">");
```
Comment (kortemik) [inline src/main/java/com/teragrep/rlo_12/FileStatusManager.java:154]: change to formatter pattern

File: src/main/java/com/teragrep/rlo_12/FileStatusManager.java [hunk 12]
```diff
@@ -132,20 +145,28 @@ public void run() {
...
+                                    }
                                     pathCurrentStateMap.put(changedFile.getPath(), MonitoredFile.Status.MODIFIED_SYNCHRONIZING);
                                     threadPoolExecutor.execute(new FileTask(transferQueue, new MonitoredFile(changedFile.getPath(), MonitoredFile.Status.SYNC_RECREATED), consumerSupplier.get()));
                                     break;
                                 case SYNC_MODIFIED:
-                                    LOGGER.trace("Path <[" + changedFile.getPath() + "]> state changed from <" + currentState + "> to <" + MonitoredFile.Status.MODIFIED_SYNCHRONIZING + "> because <" + changedFile.getStatus() + ">");
+                                    if(LOGGER.isTraceEnabled()) {
+                                        LOGGER.trace("Path <[" + changedFile.getPath() + "]> state changed from <" + currentState + "> to <" + MonitoredFile.Status.MODIFIED_SYNCHRONIZING + "> because <" + changedFile.getStatus() + ">");
```
Comment (kortemik) [inline src/main/java/com/teragrep/rlo_12/FileStatusManager.java:161]: change to formatter pattern

File: src/main/java/com/teragrep/rlo_12/FileStatusManager.java [hunk 13]
```diff
@@ -132,20 +145,28 @@ public void run() {
...
+                                    }
                                     pathCurrentStateMap.put(changedFile.getPath(), MonitoredFile.Status.MODIFIED_SYNCHRONIZING);
                                     threadPoolExecutor.execute(new FileTask(transferQueue, new MonitoredFile(changedFile.getPath(), MonitoredFile.Status.SYNC_MODIFIED), consumerSupplier.get()));
                                     break;
                                 case SYNC_DELETED:
-                                    LOGGER.trace("Path <[" + changedFile.getPath() + "]> state changed from <" + currentState + "> to <" + MonitoredFile.Status.DELETE_SYNCHRONIZING + "> because <" + changedFile.getStatus() + ">");
+                                    if(LOGGER.isTraceEnabled()) {
+                                        LOGGER.trace("Path <[" + changedFile.getPath() + "]> state changed from <" + currentState + "> to <" + MonitoredFile.Status.DELETE_SYNCHRONIZING + "> because <" + changedFile.getStatus() + ">");
```
Comment (kortemik) [inline src/main/java/com/teragrep/rlo_12/FileStatusManager.java:168]: change to formatter pattern

File: src/main/java/com/teragrep/rlo_12/FileStatusManager.java [hunk 14]
```diff
@@ -162,11 +183,15 @@ public void run() {
                             switch (currentState) {
                                 case SYNC_DELETED:
                                 case DELETE_SYNCHRONIZING:
-                                    LOGGER.trace("Path <[" + changedFile.getPath() + "]> state changed from <" + currentState + "> to <NONE> because <" + changedFile.getStatus() + ">");
+                                    if(LOGGER.isTraceEnabled()) {
+                                        LOGGER.trace("Path <[" + changedFile.getPath() + "]> state changed from <" + currentState + "> to <NONE> because <" + changedFile.getStatus() + ">");
```
Comment (kortemik) [inline src/main/java/com/teragrep/rlo_12/FileStatusManager.java:187]: change to formatter pattern

File: src/main/java/com/teragrep/rlo_12/FileStatusManager.java [hunk 15]
```diff
@@ -162,11 +183,15 @@ public void run() {
...
+                                    }
                                     pathCurrentStateMap.remove(changedFile.getPath());
                                     break;
                                 case SYNC_RECREATED:
-                                    LOGGER.trace("Path <[" + changedFile.getPath() + "]> state changed from <" + currentState + "> to <" + MonitoredFile.Status.SYNC_RECREATED + "> because <" + changedFile.getStatus() + ">");
+                                    if(LOGGER.isTraceEnabled()) {
+                                        LOGGER.trace("Path <[" + changedFile.getPath() + "]> state changed from <" + currentState + "> to <" + MonitoredFile.Status.SYNC_RECREATED + "> because <" + changedFile.getStatus() + ">");
```
Comment (kortemik) [inline src/main/java/com/teragrep/rlo_12/FileStatusManager.java:193]: change to formatter pattern

File: src/main/java/com/teragrep/rlo_12/FileTask.java [hunk 16]
```diff
@@ -53,8 +53,9 @@ class FileTask implements Runnable {
 
     @Override
     public void run() {
-        LOGGER.trace("FileTask on path <[" + monitoredFile.getPath() + "]> for <" + monitoredFile.getStatus() + ">");
-
+        if(LOGGER.isTraceEnabled()) {
+            LOGGER.trace("FileTask on path <[" + monitoredFile.getPath() + "]> for <" + monitoredFile.getStatus() + ">");
```
Comment (kortemik) [inline src/main/java/com/teragrep/rlo_12/FileTask.java:57]: change to formatter pattern

File: src/main/java/com/teragrep/rlo_12/FileTask.java [hunk 17]
```diff
@@ -73,7 +74,9 @@ public void run() {
         }
         try {
             resultQueue.transfer(result);
-            LOGGER.trace("Successfully processed <[" + result.getPath() + "]> returned with status <"+result.getStatus()+">");
+            if(LOGGER.isTraceEnabled()) {
+                LOGGER.trace("Successfully processed <[" + result.getPath() + "]> returned with status <" + result.getStatus() + ">");
```
Comment (kortemik) [inline src/main/java/com/teragrep/rlo_12/FileTask.java:78]: change to formatter pattern

**Comment by StrongestNumber9** [comment]
Implemented the change requests

**Review by kortemik — APPROVED** [review]

**Pull request merged**